staging/src/k8s.io/cri-client/pkg/remote_image.go

```diff
@@ -30,11 +30,11 @@ import (
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 	tracing "k8s.io/component-base/tracing"
+	"k8s.io/cri-client/pkg/util"
 	"k8s.io/klog/v2"
 
 	internalapi "k8s.io/cri-api/pkg/apis"
 	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
-	"k8s.io/kubernetes/pkg/kubelet/util"
 )
 
 // remoteImageService is a gRPC implementation of internalapi.ImageManagerService.
```

Comment:
Are we keeping the error logging from inside the library code (`klog.ErrorS`)? Typically libraries returns descriptive error object and let caller decide how to log it.

Reply:
The problem is that klog.Logger is not capable of logging structured via `InfoS` (only `Info`). Is it okay to drop that requirement? Then I'd say we can pass a `*klog.Logger` to the services which can be `nil` as well: https://github.com/kubernetes/kubernetes/pull/124741

Reply:
Solved.